src/toolkit/Community.VisualStudio.Toolkit.Shared/Helpers/SingleFileGeneratorTypes.cs

```diff
@@ -0,0 +1,27 @@
+namespace Community.VisualStudio.Toolkit
+{
+    /// <summary>
+    /// SingleFileGenerators rely on Language GUIDs rather than ProjectType GUIDs. <br/>
+    /// This class contains known types for easier registration of SingleFileGenerators.
+    /// </summary>
+    /// <remarks>
+    /// <see href="https://docs.microsoft.com/en-us/dotnet/api/microsoft.visualstudio.shell.interop.ivssinglefilegenerator?view=visualstudiosdk-2019"/>
+    /// <para> Example class Attribute Usage: <br/>
+    /// [Guid("Guid-For-Your-IVsSingleFileGenerator")] <br/>
+    /// [ComVisible(true)] <br/>
+    /// [ProvideObject(typeof(YourCustomToolClass))] <br/>
+    /// [CodeGeneratorRegistration(typeof(YourCustomToolClass), "NameOfCustomTool", SingleFileGeneratorTypes.CSHARP, GeneratesDesignTimeSource = true, GeneratorRegKeyName = "CustomTool for C#")] <br/>
+    /// </remarks>
+    public static class SingleFileGeneratorTypes
```

Comment:
If these indeed are based on a 'Language' then I might suggest naming this something like `LanguageTypes`.  That being said, I don't know enough about these GUID's and where they all might be used so perhaps this name makes sense as-is.

Reply:
So, the reason I named it like that is because the only reference I've been able to find for those GUIDS was specifically for SingleFileGenerators, such as the MSDataSetGenerator microsoft provides for XSD files. 

In my original issue, #221 , I had also shown where I got those GUIDs from: the 'Generators' registry key. 

Now, it _may_ make sense for it to be renamed to LanguageTypes, I don't disagree there. But I kinda figured that that would already be synonymous to the ProjectType constant class, since your project is compiling a specific language anyway.